src/Evolution/Systems/Burgers/BoundaryConditions/Periodic.hpp

```diff
@@ -0,0 +1,50 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#pragma once
+
+#include <memory>
+#include <pup.h>
+
+#include "Domain/BoundaryConditions/Periodic.hpp"
+#include "Evolution/Systems/Burgers/BoundaryConditions/BoundaryCondition.hpp"
+#include "Options/Options.hpp"
+#include "Parallel/CharmPupable.hpp"
+#include "Utilities/Gsl.hpp"
+#include "Utilities/TMPL.hpp"
+
+namespace Burgers::BoundaryConditions {
+/*!
+ * \brief Periodic boundary conditions.
+ *
+ * \note Not all domain creators will allow you to specify periodic boundary
+ * conditions since they may not make sense.
+ */
+class Periodic final : public BoundaryCondition,
```

Comment:
Something that might be possible here is to instead use CRTP and have a generic Periodic class that is just templated on the system-specific base class. That means all that anyone would need to do to handle periodic BCs is add `domain::BoundaryConditions::Periodic<Burgers::BoundaryConditions::BoundaryCondition>` to the list of allowed boundary conditions. This would remove a lot of the repeated code, and would also translate nicely to a `None` boundary condition.